restapi/src/main/java/io/stargate/web/resources/Converters.java

```diff
@@ -1031,4 +1036,126 @@ public static String maybeQuote(String text) {
+      case INET:
+        return Column.Type.Inet;
+      case TIMEUUID:
+        return Column.Type.Timeuuid;
+      case TIMESTAMP:
+        return Column.Type.Timestamp;
+      case BIGINT:
+        return Column.Type.Bigint;
+      case TIME:
+        return Column.Type.Time;
+      case DURATION:
+        return Column.Type.Duration;
+      case VARINT:
+        return Column.Type.Varint;
+      case UUID:
+        return Column.Type.Uuid;
+      case BOOLEAN:
+        return Column.Type.Boolean;
+      case TINYINT:
+        return Column.Type.Tinyint;
+      case SMALLINT:
+        return Column.Type.Smallint;
+      case ASCII:
+        return Column.Type.Ascii;
+      case DECIMAL:
+        return Column.Type.Decimal;
+      case BLOB:
+        return Column.Type.Blob;
+      case VARCHAR:
+      case TEXT:
+        return Column.Type.Text;
+      case DOUBLE:
+        return Column.Type.Double;
+      case COUNTER:
+        return Column.Type.Counter;
+      case DATE:
+        return Column.Type.Date;
+      case FLOAT:
+        return Column.Type.Float;
+      case LIST:
+        if (info == null) {
+          throw new IllegalArgumentException(
+              "List cqlType should contain an 'info' field specifying the sub cqlType");
+        }
+        if (typeParams == null || typeParams.size() != 1) {
+          throw new IllegalArgumentException("List sub types should contain 1 item");
+        }
+        return Column.Type.List.of(decodeType(typeParams.get(0))).frozen(frozen);
+      case SET:
+        if (info == null) {
+          throw new IllegalArgumentException(
+              "Set cqlType should contain an 'info' field specifying the sub cqlType");
+        }
+        if (typeParams == null || typeParams.size() != 1) {
+          throw new IllegalArgumentException("Set sub types should contain 1 item");
+        }
+        return Column.Type.Set.of(decodeType(typeParams.get(0))).frozen(frozen);
+      case MAP:
+        if (info == null) {
+          throw new IllegalArgumentException(
+              "Map cqlType should contain an 'info' field specifying the sub types");
+        }
+        if (typeParams == null || typeParams.size() != 2) {
+          throw new IllegalArgumentException("Map sub types should contain 2 items");
+        }
+        return Column.Type.Map.of(decodeType(typeParams.get(0)), decodeType(typeParams.get(1)))
+            .frozen(frozen);
+      case UDT:
+        if (info == null || Strings.isNullOrEmpty(info.getName())) {
+          throw new IllegalArgumentException(
+              "UDT cqlType should contain an 'info' field specifying the UDT name");
+        }
+        return UserDefinedType.reference(udtType.getName()).frozen(frozen);
+      case TUPLE:
+        if (info == null) {
+          throw new IllegalArgumentException(
+              "TUPLE cqlType should contain an 'info' field specifying the sub types");
+        }
+        if (typeParams == null || typeParams.isEmpty()) {
```

Comment:
Why is this checking `isEmpty` but above it's `typeParams.size() != 1`?

Reply:
Because tuple may have many sub types, but list, for example, can only have one subtype.